packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,440 @@
+
+      await execa(
+        'yarn',
+        ['config', 'set', 'npmRegistryServer', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa(
+        'yarn',
+        ['config', 'set', 'unsafeHttpWhitelist', 'localhost'],
+        execaOptions
+      );
+    } else {
+      await execa(
+        'yarn',
+        ['config', 'set', 'registry', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa('yarn', ['config', 'get', 'registry'], execaOptions);
+    }
+    await execa('yarn', ['cache', 'clean'], execaOptions);
+  } else if (packageManagerExecutable === 'pnpm') {
+    await execa(packageManagerExecutable, ['--version']);
+    await execa(packageManagerExecutable, [
+      'config',
+      'set',
+      'registry',
+      'http://localhost:4873',
+    ]);
+    await execa(packageManagerExecutable, ['config', 'get', 'registry']);
+  }
+};
+
+void describe('amplify', { concurrency: concurrency }, () => {
+  const { PACKAGE_MANAGER_EXECUTABLE = 'npm' } = process.env;
+  const packageManagerExecutable = PACKAGE_MANAGER_EXECUTABLE.startsWith('yarn')
+    ? 'yarn'
+    : PACKAGE_MANAGER_EXECUTABLE;
+
+  before(async () => {
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
+    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+
+    // install package manager
+    if (PACKAGE_MANAGER_EXECUTABLE === 'yarn-classic') {
+      await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+    } else if (PACKAGE_MANAGER_EXECUTABLE === 'pnpm') {
+      await execa('npm', ['install', '-g', PACKAGE_MANAGER_EXECUTABLE], {
+        stdio: 'inherit',
+      });
+    }
+
+    // nuke the npx cache to ensure we are installing packages from the npm proxy
+    if (PACKAGE_MANAGER_EXECUTABLE !== 'yarn-modern') {
+      await packageManagerSetup(
+        PACKAGE_MANAGER_EXECUTABLE as PackageManagerExecutable
+      );
+    }
+
+    // Force 'create-amplify' installation in npx cache by executing help command
+    // before tests run. Otherwise, installing 'create-amplify' concurrently
+    // may lead to race conditions and corrupted npx cache.
+    await execa(
+      packageManagerExecutable,
+      [
+        'create',
+        'amplify',
+        ...(PACKAGE_MANAGER_EXECUTABLE === 'yarn-modern'
+          ? []
+          : ['--yes', '--']),
+        '--help',
+      ],
+      {
+        // Command must run outside of 'amplify-backend' workspace.
+        cwd: os.homedir(),
+        stdio: 'inherit',
+      }
+    );
+  });
+
+  after(async () => {
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  const initialStates = ['empty', 'module', 'commonjs'] as const;
+
+  initialStates.forEach((initialState) => {
+    void describe('installs expected packages and scaffolds expected files', () => {
```

Comment:
we shouldn't repeat tests from create amplify if we're creating new file.
i.e. we should start with single test that does full create-amplify -> deployment flow here.

which means that we should:
1. remove initialStates, and just start from `empty`
2. remove most of assertions that repeat create amplify test, i.e. it should be similar to https://github.com/aws-amplify/amplify-backend/blob/main/packages/integration-tests/src/test-live-dependency-health-checks/health_checks.test.ts